python/doc/examples/calibration/bayesian_calibration/plot_imh_python_distribution.py

```diff
@@ -1,100 +1,327 @@
+            return 1.0
+        return ot.GaussLegendre([32]).integrate(
+            self.PDF_, ot.Interval(self.rangeLow_, X[0])
+        )[0]
+
+    def getRealization(self):
+        return self.sampler_.getRealization()
+
+    def getSample(self, n):
+        return self.sampler_.getSample(n)
+
+
+# %%
+# We create the new distribution that uses the Ratio of Uniforms sampler:
+newDist_RoU = ot.Distribution(NewDistribution_RoU())
+
+# %%
+# We compare the time for sampling the distribution with the Ratio of Uniforms algorithm to
+# the time with the generic sampler. The Ratio of Uniforms algorithm proves to be much more rapid
+# than the generic method.
+sizeRoU = 10000
+sizeGeneric = 100
+
+t0 = monotonic()
+sample_RoU = newDist_RoU.getSample(sizeRoU)
+t1 = monotonic()
+sample_generic = newDist_generic.getSample(sizeGeneric)
+t2 = monotonic()
+
+print("Ratio of Uniforms speed = ", sizeRoU / (t1 - t0), "rng/s")
+print("Generic algorithm speed = ", sizeGeneric / (t2 - t1), "rng/s")
+
+# %%
+# Case 2: Direct use the Ratio of Uniforms algorithm
+# --------------------------------------------------
+# In that case, we want to use the :class:`~openturns.experimental.RatioOfUniforms` algorithm to
+# sample :math:`p`. We need to compute the function :math:`\log f` and its range.
+#
+# We create the function :math:`\log f` and the :class:`~openturns.experimental.RatioOfUniforms`:
+log_UnscaledPDF = ot.ComposedFunction(ot.SymbolicFunction("x", "log(x)"), f)
+ratio_algo = otexp.RatioOfUniforms(log_UnscaledPDF, range_PDF)
+
+# %%
+# We can draw the exact PDF and the PDF estimated from a sample generated by the Ratio of Uniforms algorithm.
+
+# %%
+size = 100000
+sample = ratio_algo.getSample(size)
+
+ks_algo = ot.KernelSmoothing()
+ks_algo.setBoundaryCorrection(True)
+ks_algo.setLowerBound(lower_bound)
+ks_algo.setUpperBound(upper_bound)
+ks_pdf = ks_algo.build(sample)
+
+g = ks_pdf.drawPDF(-0.5, upper_bound * 1.1, 1001)
+g.add(draw_exact)
+g.setLegends(["RoU size = " + str(size), "exact PDF"])
+g.setLegendPosition("topright")
+g.setTitle("Ratio of Uniforms (RoU) ")
+g.setXTitle("x")
+view = otv.View(g)
+
+# %%
+# In order to facilitate the comparison with the use of Metropolis-Hastings
+# based algorithms, we generate
+# a sample of the same size and we draw the estimated PDF.
+size2 = 500
+sample = ratio_algo.getSample(size2)
+
+ks_algo = ot.KernelSmoothing()
+ks_algo.setBoundaryCorrection(True)
+ks_algo.setLowerBound(lower_bound)
+ks_algo.setUpperBound(upper_bound)
+ks_pdf_RoU = ks_algo.build(sample)
+
+draw_RoU = ks_pdf_RoU.drawPDF(-0.5, upper_bound * 1.1, 1001).getDrawable(0)
+draw_RoU.setLineWidth(2)
+g.add(draw_RoU)
+g.setLegends(
+    ["RoU size = " + str(size), "exact PDF", "RoU (size = " + str(size2) + ")"]
+)
+g.setTitle("Ratio of Uniforms")
+view = otv.View(g)
 
 # %%
-# Independent Metropolis-Hastings
-# -------------------------------
+# By default, the parameter :math:`r=1`. We can get the associated acceptance ratio:
```

Comment:
I would add a "see the documentation for more information on the parameter purpose"

Reply:
ok. I don't know how to create  link to another example....